src/test/java/net/snowflake/client/jdbc/PreparedMultiStmtIT.java

```diff
@@ -5,39 +5,49 @@
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.Assert.assertTrue;
 
-import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
 import net.snowflake.client.category.TestCategoryStatement;
 import org.junit.Assert;
+import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
 @Category(TestCategoryStatement.class)
-public class PreparedMultiStmtIT extends BaseJDBCTest {
+public class PreparedMultiStmtIT extends BaseJDBCWithSharedConnectionIT {
 
   protected static String queryResultFormat = "json";
 
-  public static Connection getConnection() throws SQLException {
-    Connection conn = BaseJDBCTest.getConnection();
-    try (Statement stmt = conn.createStatement()) {
+  private static SnowflakeConnectionV1 sfConnectionV1;
+
+  @BeforeClass
+  public static void setUpPreparedStatementConnection() throws SQLException {
+    if (!connection.isClosed()) {
```

Comment:
how can it be possible to have closed connection here? it should be renew by the base class before tests in this class?

Reply:
I actually was wondering about this as well because when I ran this locally it never failed, but running in github it was constantly showing "connection closed" errors. When I added println statements to see where the connection was getting closed while running on github it looked like PreparedMultiStmtArrowIT (which extends PreparedMultiStmtIT and essentially just runs the same tests with arrow as the result format) would run first and then close that shared connection, and for some reason the "BeforeClass" from BaseJDBCWithSharedConnectionIT wasn't running again for the second test suite.

Reply:
Indeed it may happen for parameterized tests
then we can organize it in a different way in tests and have e.g. connections with specific result format set upfront

e.g. for parameterized tests we can recreate the connection in the constructor when necessary and it may involve not using the same base class that you have created before

I addressed it in the not merged PR https://github.com/snowflakedb/snowflake-jdbc/pull/1754/files#diff-b01b6456eab10190c84533701ca0d1a477f5463c060522920c63fa25ab6fc765, you can take an inspiration from this PR

Reply:
It's not actually running as a true parameterized test though. PreparedMultiStmtArrowIT simply extends PreparedMultiStmtIT and calls the super constructor passing in the "arrow" query format (https://github.com/snowflakedb/snowflake-jdbc/blob/1d5b6951066a2e40fe466475cf4539eddac3f29f/src/test/java/net/snowflake/client/jdbc/PreparedMultiStmtArrowIT.java#L7) 

I'm thinking if we want to change this, it might be better to simplify and refactor PreparedMultiStmtIT to actually be parameterized and run both json and arrow tests and to delete PreparedMultiStmtArrowIT, rather than to introduce an additional base class for the connection sharing. Unless there is some reason we want to keep the arrow tests as a distinct class?

Reply:
it's a great idea to go fully with parameterized tests. The reason why it's split may be will to use TestCategoryArrow, but it should not be the blocker - please rewrite the tests here.
The same case applies in MultiStatementArrowIT, ResultSetArrowIT, ResultSetArrowLatestIT, PreparedStatementArrow1IT, PreparedStatementArrow1LatestIT, PreparedStatementArrow2IT, StatementArrowIT

Based on that it is possible that we will need to think about another categories and it will affect distribution in https://github.com/snowflakedb/snowflake-jdbc/pull/1735

Reply:
I've made the change for PreparedStatementIT here, but some of these other cases were covered in other PR's. I think it would be best to address all of these cases as one separate PR.